app/controllers/news_controller.rb

```diff
@@ -1,22 +1,32 @@
 class NewsController < ApplicationController
   include NodesHelper
   include NewsHelper
-  
-  before_action :set_section, only: [:show]
+
+  before_action :set_section, only: [:index, :show]
+  before_action :set_filters, only: [:index]
 
   decorates_assigned :node
   decorates_assigned :articles
 
+
   def index
     # TODO: this will be refactored out once node-hierarchy nav is implemented
     # For now, this is the quickest way to get the same outcome
     if params[:section].present?
-      set_section
       set_menu_nodes
-      @articles = news_articles section: @section
+      @filters = [@section]
+      @articles = news_articles
     else
       @articles = news_articles
+
+      if @filters.present?
+        @articles = @articles
+                        .includes(:sections)
+                        .references(:sections)
+                        .where('sections.id IN (?)', @filters.collect(&:id))
+      end
```

Comment:
Would it suffice to rewrite the declaration of line 23 with

`@articles = news_articles.`

This would preload the associations via `news_helper.rb`.